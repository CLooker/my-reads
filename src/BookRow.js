import React from 'react';
import './App.css';

const BookRow = function(props) {
    return (
      <div className="list-books">
        <div className="list-books-title">
          <h1>{props.myReads}</h1>
        </div>
        <div className="list-books-content">
          <div className="bookshelf">
            <h2 className="bookshelf-title">{props.shelfTitle}</h2>
            <div className="bookshelf-books">
              <ol className="books-grid">
                {props.shelf.map(book => {
                  return (
                    <li key={book.id}>
                      <div className="book">
                        <div className="book-top">
                          <div className="book-cover"
                            style={{
                              width: 128,
                              height: 193,
                              backgroundImage:
                                (`url(
                                    ${(book.imageLinks && book.imageLinks.thumbnail) ||
                                    ("http://via.placeholder.com/128x193")}
                                  )`
                                )
                            }}>
                          </div>
                          <div className="book-shelf-changer">
                            <select
                              onChange={props.changeBookshelf}
                              value={props.selectedValue}
                              data={book.id}
                            >
                              <option value="none" disabled>Move to...</option>
                              {/*here displaySearch being true/false short-circuits
                              for some conditional rendering*/}
                              {props.displaySearch && <option value=""></option>}
                              <option value="currentlyReading">Currently Reading</option>
                              <option value="wantToRead">Want to Read</option>
                              <option value="read">Read</option>
                              {props.displaySearch || <option value="none">None</option>}
                              </select>
                          </div>
                        </div>
                        <div className="book-title">{book.title}</div>
                        <div className="book-authors">
                          {book.authors && book.authors.map((author)=> {
                            return (<div key={author + Math.random()}>{author}</div>)
                            })}
                        </div>
                      </div>
                    </li>
                  )
                })}
              </ol>
            </div>
          </div>
        </div>
      </div>
    )
}

export default BookRow;
