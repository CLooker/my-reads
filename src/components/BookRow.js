import React from 'react';
import Book from './Book';
import PropTypes from 'prop-types';

const BookRow = ({
  myReads,
  shelfTitle,
  shelf,
  changeBookshelf,
  displaySearch
}) => (
  <div className="list-books">
    <div className="list-books-title">
      <h1>{myReads}</h1>
    </div>
    <div className="list-books-content">
      <div className="bookshelf">
        <h2 className="bookshelf-title">{shelfTitle}</h2>
        <div className="bookshelf-books">
          <ol className="books-grid">
            {shelf.map(book => (
              <li key={book.id}>
                <Book
                  book={book}
                  changeBookshelf={changeBookshelf}
                  displaySearch={displaySearch}
                />
              </li>
            ))}
          </ol>
        </div>
      </div>
    </div>
  </div>
);

BookRow.propTypes = {
  myReads: PropTypes.string,
  shelfTitle: PropTypes.string.isRequired,
  shelf: PropTypes.array.isRequired,
  changeBookshelf: PropTypes.func.isRequired,
  displaySearch: PropTypes.bool
};

export default BookRow;
