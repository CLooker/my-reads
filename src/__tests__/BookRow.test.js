import React from 'react';
import { shallow } from 'enzyme';
import BookRow from '../components/BookRow';

describe('BookRow', () => {
  const componentProps = {
    shelfTitle: 'Currently Reading',
    shelf: [
      { id: '123', title: 'fake title' },
      { id: '456', title: 'fake title 2' }
    ],
    changeBookshelf: jest.fn()
  };

  let component;

  beforeEach(() => {
    component = shallow(<BookRow {...componentProps} />);
  });

  describe('initialization', () => {
    describe('code paths', () => {
      describe('`myReads` is a prop', () => {
        it('should render properly', () => {
          const localComponentProps = {
            ...componentProps,
            myReads: 'MyReads'
          };

          const localComponent = shallow(<BookRow {...localComponentProps} />);

          expect(localComponent).toMatchSnapshot();
        });
      });

      describe('`display is a truthy prop`', () => {
        it('should render properly', () => {
          const localComponentProps = {
            ...componentProps,
            displaySearch: true
          };

          const localComponent = shallow(<BookRow {...localComponentProps} />);

          expect(localComponent).toMatchSnapshot();
        });
      });

      describe('no optional props', () => {
        it('should render properly', () => {
          expect(component).toMatchSnapshot();
        });
      });
    });
  });
});
