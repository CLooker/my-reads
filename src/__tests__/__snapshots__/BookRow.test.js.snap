// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookRow initialization code paths \`display is a truthy prop\` should render properly 1`] = `
<div
  className="list-books"
>
  <div
    className="list-books-title"
  >
    <h1 />
  </div>
  <div
    className="list-books-content"
  >
    <div
      className="bookshelf"
    >
      <h2
        className="bookshelf-title"
      >
        Currently Reading
      </h2>
      <div
        className="bookshelf-books"
      >
        <ol
          className="books-grid"
        >
          <li
            key="123"
          >
            <Book
              book={
                Object {
                  "id": "123",
                  "title": "fake title",
                }
              }
              changeBookshelf={[MockFunction]}
              displaySearch={true}
            />
          </li>
          <li
            key="456"
          >
            <Book
              book={
                Object {
                  "id": "456",
                  "title": "fake title 2",
                }
              }
              changeBookshelf={[MockFunction]}
              displaySearch={true}
            />
          </li>
        </ol>
      </div>
    </div>
  </div>
</div>
`;

exports[`BookRow initialization code paths \`myReads\` is a prop should render properly 1`] = `
<div
  className="list-books"
>
  <div
    className="list-books-title"
  >
    <h1>
      MyReads
    </h1>
  </div>
  <div
    className="list-books-content"
  >
    <div
      className="bookshelf"
    >
      <h2
        className="bookshelf-title"
      >
        Currently Reading
      </h2>
      <div
        className="bookshelf-books"
      >
        <ol
          className="books-grid"
        >
          <li
            key="123"
          >
            <Book
              book={
                Object {
                  "id": "123",
                  "title": "fake title",
                }
              }
              changeBookshelf={[MockFunction]}
            />
          </li>
          <li
            key="456"
          >
            <Book
              book={
                Object {
                  "id": "456",
                  "title": "fake title 2",
                }
              }
              changeBookshelf={[MockFunction]}
            />
          </li>
        </ol>
      </div>
    </div>
  </div>
</div>
`;

exports[`BookRow initialization code paths no optional props should render properly 1`] = `
<div
  className="list-books"
>
  <div
    className="list-books-title"
  >
    <h1 />
  </div>
  <div
    className="list-books-content"
  >
    <div
      className="bookshelf"
    >
      <h2
        className="bookshelf-title"
      >
        Currently Reading
      </h2>
      <div
        className="bookshelf-books"
      >
        <ol
          className="books-grid"
        >
          <li
            key="123"
          >
            <Book
              book={
                Object {
                  "id": "123",
                  "title": "fake title",
                }
              }
              changeBookshelf={[MockFunction]}
            />
          </li>
          <li
            key="456"
          >
            <Book
              book={
                Object {
                  "id": "456",
                  "title": "fake title 2",
                }
              }
              changeBookshelf={[MockFunction]}
            />
          </li>
        </ol>
      </div>
    </div>
  </div>
</div>
`;
